t/cli/test_etcd_sync_event_handle.sh

```diff
@@ -0,0 +1,133 @@
+etcdctl --endpoints=127.0.0.1:2379 --user=root:apache-api6-sync put /apisix/routes/2 '{"uri":"/2"}' ## set incorrect configuration
+etcdctl --endpoints=127.0.0.1:2379 --user=root:apache-api6-sync put /apisix/routes/3 '{"uri":"/3","plugins":{"fault-injection":{"abort":{"http_status":204}}}}'
+etcdctl --endpoints=127.0.0.1:2379 --user=root:apache-api6-sync put /apisix/routes/4 '{"uri":"/4","plugins":{"fault-injection":{"abort":{"http_status":204}}}}'
+etcdctl --endpoints=127.0.0.1:2379 --user=root:apache-api6-sync put /apisix/routes/5 '{"uri":"/5","plugins":{"fault-injection":{"abort":{"http_status":204}}}}'
+
+# Resume APISIX synchronization by disable auth
+# Since APISIX will not be able to access etcd until authentication is disable,
+# watch will be temporarily disabled, so when authentication is disable,
+# the backlog events will be sent at once at an offset from when APISIX disconnects.
+# When APISIX resumes the connection, it still has not met its mandatory full
+# synchronization condition, so it will be "watch" that resumes, not "readdir".
+etcdctl --endpoints=127.0.0.1:2379 --user=root:apache-api6-sync auth disable
+etcdctl --endpoints=127.0.0.1:2379 user delete root
+etcdctl --endpoints=127.0.0.1:2379 role delete root
+sleep 5 # wait resync by watch
+
+# Test request
+# All but the intentionally incoming misconfigurations should be applied,
+# and non-existent routes will remain non-existent.
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/1 | grep 204 || (echo "failed: Round 2 Request 1 unexpected"; exit 1)
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/2 | grep 503 || (echo "failed: Round 2 Request 2 unexpected"; exit 1)
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/3 | grep 204 || (echo "failed: Round 2 Request 3 unexpected"; exit 1)
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/4 | grep 204 || (echo "failed: Round 2 Request 4 unexpected"; exit 1)
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/5 | grep 204 || (echo "failed: Round 2 Request 5 unexpected"; exit 1)
+curl -s -o /dev/null -w "%{http_code}" http://127.0.0.1:9080/6 | grep 404 || (echo "failed: Round 2 Request 6 unexpected"; exit 1)
+
+# Check logs
+## Case1: Ensure etcd is disconnected
+cat logs/error.log | grep "watchdir err: has no healthy etcd endpoint available" || (echo "Log case 1 unexpected"; exit 1)
+
+## Case2: Ensure events are sent in bulk after connection is restored
+## It is extracted from the structure of following type
+## result = {
+##   events = { {
+##     {
+##       kv = {
+##         key = "/apisix/routes/1",
+##         ...
+##       }
+####   }, {
+##       kv = {
+##         key = "/apisix/routes/2",
+##         ...
+##       }
+##     },
+##     ...
+##   } },
+##   header = {
+##     ...
+##   }
+## }
+## After check, it only appears when watch recovers and returns events in bulk.
+cat logs/error.log | grep "}, {" || (echo "failed: Log case 2 unexpected"; exit 1)
```

Comment:
assert `failed to check item data` error log too.

Reply:
test needs to fail if expectations are not met `||` gives a exit code 0 which is not detected as a failure by the cli testing suite.

Reply:
@shreemaan-abhishek  Actually, the error doesn't cause the test suite to exit either, I'm not sure what's happening. But I've checked the log manually and it has no errors.

Reply:
@nic-6443 Good catch! Updated.